src/ChessBoard.scala

```diff
@@ -0,0 +1,153 @@
+    }
+
+    def checkDiagonals(rowI: Int, row:Int, colI:Int, col:Int): Boolean ={
+      (0 to newRow.size).exists { i =>
+        ((rowI == (row + i) && colI == (col + i)) || (rowI == (row - i) && colI == (col + i)) ||
+          (rowI == (row - i) && colI == (col - i)) || (rowI == (row + i) && colI == (col - i))) && (rowI != row && colI != col)
+      }
+
+    }
+
+    def checkRow(rowI: Int, row:Int, colI:Int, col:Int): Boolean ={
+      rowI == row && colI != col
+    }
+
+    def checkCol(rowI: Int, row:Int, colI:Int, col:Int): Boolean ={
+      rowI != row && colI == col
+    }
+
+    def markInvalidSquares(b: Board): Board = {
+      var markedBoard = b
+      b.zipWithIndex.flatMap { case (rowL, rowI) =>
+        rowL.zipWithIndex.map { case (colL, colI) =>
+          if (isInvalidSquare(rowI, row, colI, col)) {
+            markedBoard = markXs(markedBoard, rowI, colI)
+          }
+        }
+      }
+      markedBoard
+    }
+
+    val boardsWMarks = markInvalidSquares(newBoard)
+
+    boardsWMarks
+  }
+
+  def countQueens(b: Board): Int = {
+    var countQs = 0
+    b.zipWithIndex.flatMap { case (rowL, rowI) =>
+      rowL.zipWithIndex.map { case (colL, colI) =>
+        if(colL == "Q"){
+          countQs = countQs + 1
+        }
+      }
+    }
+    countQs
+  }
+
+  def placeNextQueen(groupOfBoards: Seq[Board], qCount: Int): Seq[Board] = {
+    groupOfBoards.flatMap { tmpBoard =>
+      tmpBoard.zipWithIndex.flatMap { case (row, rowI) =>
+        row.zipWithIndex.filter { case(col,colI) =>
+          // check if safe
+          isSafe(tmpBoard,rowI,colI)
+        }
+          .map { case (col, colI) =>
+
+            val newBoard = placeQueen(tmpBoard,rowI,colI)
+            // newnewnewborad = to place 3rd queen, and so on
+            newBoard
+          }
+      }
+
+    }.filter { tmpBoard =>
+      (countQueens(tmpBoard) == qCount)
+    }.distinct
+  }
+
+  // down the road - we need to use is safe and this function recurisvesly
+  def placeQueens(n: Int): Seq[Board] = {
+    // create board - size n
+
+    var qCount = 1
+
+    var boardsWithQueens = placeNextQueen(Seq(createBoard(n)),1)
+
+    while(qCount < n){
+      qCount = qCount + 1
+      boardsWithQueens = placeNextQueen(boardsWithQueens,qCount)
+    }
+    // any boards left? when n? - if so then we have valid boards
+
+    boardsWithQueens.distinct
+
+  }
+
+  def isSafe(board: Board, row: Int, col: Int): Boolean = {
+    //how to determine if safe?
+    var rowCounter = -1
+    var colCounter = -1
+    var checkBoard = board.exists { boardRow =>
+      rowCounter = rowCounter + 1
+      var checkCol = boardRow.exists { boardSquare =>
+        colCounter = colCounter + 1
+        ((boardSquare == "X" || boardSquare == "Q") & ((rowCounter == row) & (colCounter == col)))
+      }
+      checkCol
+    }
+    !checkBoard
+  }
+
+  // just one queen at the moment
+  //
+  val solvedBoards = placeQueens(6)
```

Comment:
As a finishing touch to this code I think it would be cool to add a command line prompt. You could run this code from the terminal and have something like this print out:
> Welcome to the N-Queens program! Please enter an number (N) to specify the chess board size and number of queens to place
> \> 6
> Solving N-Queens with N of 6...
> Solved boards:
> ....

This would also involve adding some error handling in the case a user tries to enter a String or decimal number or something like that